packages/app/components/TokenIcon.tsx

```diff
@@ -0,0 +1,72 @@
+import { isAddress } from "viem";
+import { cva } from "class-variance-authority";
+import Image from "next/image";
+import { useNetwork } from "wagmi";
+import { twMerge } from "tailwind-merge";
+import { useTokenListContext } from "@/context/TokenListContext";
+import { Token } from "@/models/token/types";
+
+interface TokenIconProps {
+  token: Token;
+  className?: string;
+  size?: "lg" | "md" | "sm" | "xs";
+}
+
+export const TokenIcon = ({ token, className, size }: TokenIconProps) => {
+  const { tokenList } = useTokenListContext();
+
+  function getTokenFromList(tokenAddress: string) {
+    return tokenList.find(
+      (el) => el.address.toUpperCase() === tokenAddress.toUpperCase()
+    );
+  }
```

Comment:
Should this be inside the context?

Reply:
Yeah probably is cleaner. I moved the other function as well.